workflows/metatdenovo.nf

```diff
@@ -73,17 +144,373 @@ workflow METATDENOVO {
+        .set { ch_collect_feature }
+
+    COLLECT_FEATURECOUNTS ( ch_collect_feature )
+    ch_versions           = ch_versions.mix(COLLECT_FEATURECOUNTS.out.versions)
+    ch_fcs_for_stats      = COLLECT_FEATURECOUNTS.out.counts.collect { meta, tsv -> tsv }.map { [ it ] }
+    ch_fcs_for_summary    = COLLECT_FEATURECOUNTS.out.counts.map { meta, tsv -> tsv }
+    ch_collect_stats
+        .combine(ch_fcs_for_stats)
+        .set { ch_collect_stats }
+
+    //
+    // SUBWORKFLOW: run eggnog_mapper on the ORF-called amino acid sequences
+    //
+    if ( ! params.skip_eggnog ) {
+        EGGNOG(params.eggnog_dbpath, ch_protein, ch_fcs_for_summary )
+        ch_versions = ch_versions.mix(EGGNOG.out.versions)
+        ch_merge_tables = EGGNOG.out.sumtable
+    } else {
+        ch_protein
+            .map { meta, protein -> [ meta, [] ] }
+            .set { ch_merge_tables }
+    }
+
+
+    //
+    // SUBWORKFLOW: run kofamscan on the ORF-called amino acid sequences
+    //
+    if( !params.skip_kofamscan ) {
+        ch_protein
+            .map { meta, protein -> [ meta, protein ] }
+            .set { ch_kofamscan }
+        KOFAMSCAN( ch_kofamscan, ch_fcs_for_summary)
+        ch_versions = ch_versions.mix(KOFAMSCAN.out.versions)
+        ch_kofamscan_summary = KOFAMSCAN.out.kofamscan_summary.collect().map { meta, tsv -> tsv }
+        ch_merge_tables
+            .combine( ch_kofamscan_summary )
+            .set { ch_merge_tables }
+    } else {
+        ch_merge_tables
+            .map { meta, tsv -> [ meta, tsv, [] ] }
+            .set { ch_merge_tables }
+    }
+
+    // set up contig channel to use in CAT and TransRate
+    UNPIGZ_CONTIGS(ch_assembly_contigs)
+    ch_unzipped_contigs = UNPIGZ_CONTIGS.out.unzipped
+    ch_versions = ch_versions.mix(UNPIGZ_CONTIGS.out.versions)
+
+    //
+    // MODULE: Use TransRate to judge assembly quality, piped into MultiQC
+    //
+    TRANSRATE(ch_unzipped_contigs)
+    ch_versions = ch_versions.mix(TRANSRATE.out.versions)
+
+    //
+    // SUBWORKFLOW: Eukulele
+    //
+    ch_eukulele_db = Channel.empty()
+    if( !params.skip_eukulele){
+        // Create a channel for EUKulele either with a named database or not. The latter means a user-provided database in a directory.
+        if ( params.eukulele_db ) {
+            Channel
+                .of ( params.eukulele_db.split(',') )
+                .map { [ it, file(params.eukulele_dbpath) ] }
+                .set { ch_eukulele_db }
+        } else {
+            Channel.fromPath(params.eukulele_dbpath, checkIfExists: true)
+                .map { [ [], it ] }
+                .set { ch_eukulele_db }
+        }
+        ch_protein
+            .map { meta, protein -> [ [ id:"${meta.id}" ], protein ] }
+            .combine( ch_eukulele_db )
+            .set { ch_eukulele }
+        SUB_EUKULELE( ch_eukulele, ch_fcs_for_summary )
+        ch_taxonomy_summary = SUB_EUKULELE.out.taxonomy_summary.collect().map { meta, tsv -> tsv }
```

Comment:
Check what you're getting here. You've got the same construct `.collect().map { meta, tsv -> tsv }`